src/components/single-create.ts

```diff
@@ -88,51 +88,69 @@ export default class SingleCreate {
-    mapElement.addEventListener('pointerdown', () => {
-      startTime = new Date().getTime();
-    });
-    mapElement.addEventListener('pointerup', () => {
-      endTime = new Date().getTime();
-      longpress = endTime - startTime < longClickDuration ? false : true;
+    mapElement.addEventListener('mousedown', (e) => {
+      move = 0;
+      this.clearCreationTimeout(timeout);
+      timeout = setTimeout(() => {
+        this.reqiestElementCreation(e.pageX, e.pageY, map, mapElement);
+      }, longClickDuration)
     });
 
-    map.on('click',  (e) => {
-      if (longpress) {
-        const geomPoint = new Point(e.coordinate);
-        const feature = new Feature({
-          geometry: geomPoint,
-        });
-        useStore().setSelectedFeature(feature)
-        GeocityEvent.sendEvent('icon-created', undefined);
-      }
+    mapElement.addEventListener('mousemove', () =>  {
+      move++;
+      this.moveAnalyzer(timeout, move);
+  });
+
+    mapElement.addEventListener('mouseup', () => {
+      move = 0;
+      this.clearCreationTimeout(timeout);
     });
 
     // Mobile device. 
     // Using the map div because openlayers object doesn't support the touch event. But the div yes.
-    mapElement.addEventListener('touchstart', () => {
-      startTime = new Date().getTime();
+    mapElement.addEventListener('touchstart', (e) => {
+      move = 0;
+      this.clearCreationTimeout(timeout);
+      timeout = setTimeout(() => {
+        this.reqiestElementCreation(e.changedTouches[0].pageX, e.changedTouches[0].pageY, map, mapElement);
+      }, longClickDuration)
+    });
+
+    mapElement.addEventListener('touchmove', () =>  {
+      move++;
+      this.moveAnalyzer(timeout, move);
     });
 
-    mapElement.addEventListener('touchend', (e) => {      
-      endTime = new Date().getTime();
-      if (endTime - startTime > longClickDuration) {
+    mapElement.addEventListener('touchend', () => {
+      move = 0;
+      this.clearCreationTimeout(timeout);
+    });
+  }
+
+  reqiestElementCreation(x: number, y: number, map: Map, mapElement: HTMLDivElement) {
         // To have the coordinate, we use the pixel position and the map position to find the exact pixel in the window.
         // Then use map pixel converter
-        const coordiante = map.getCoordinateFromPixel([e.changedTouches[0].pageX - mapElement.offsetLeft, e.changedTouches[0].pageY - mapElement.offsetTop]);
+        const coordiante = map.getCoordinateFromPixel([x - mapElement.offsetLeft, y - mapElement.offsetTop]);
         const geomPoint = new Point(coordiante);
         const feature = new Feature({
           geometry: geomPoint,
         });
         useStore().setSelectedFeature(feature)
         GeocityEvent.sendEvent('icon-created', undefined);
-      }
-    });
+  }
+
+  moveAnalyzer(timeout: string | number | NodeJS.Timeout | undefined, move: number) {
+    if (move > 20) this.clearCreationTimeout(timeout);
```

Comment:
why 20? does not seem really robust after testing. I would always clear the timeout after a move. At least on desktop mouse should not move on a long press, to check on touch

Reply:
Now, it's a pixel base move